### ISSUE TITLE
Add named tuple constructors for components

### ISSUE THREAD

**Issue by BSnelling** [invenia/FullNetworkSystems.jl#7]
Some of the static components defined here have lots of fields, which makes constructors hard to follow.  We could add named tuple constructors for these components so the user can do e.g.

```
br1 = Branch(;
        name="1",
        to_bus="A",
        from_bus="B",
        rate_a=10.0,
        rate_b=10.0,
        is_monitored=true,
        break_points=(100.0, 102.0),
        penalties=(5.0, 6.0),
        resistance=1.0,
        reactance=1.0
)
```

**Comment by raphaelsaavedra** [comment]
This can be done by using `Base.@kwdef` on component constructors

**Comment by nickrobinson251** [comment]
yeah, i think this would be a good addition

**Comment by nickrobinson251** [comment]
closed by #19
